probe-rs-debug/src/unit_info.rs

```diff
@@ -2029,15 +2063,13 @@ impl UnitInfo {
         // Address Pointer Conditions (any of):
         // 1. Variable names that start with '*' (e.g '*__0), AND the variable is a variant of the parent.
         // 2. Pointer names that start with '*' (e.g. '*const u8')
-        // 3. Pointers to base types (includes &str types)
-        // 4. Pointers to variable names that start with `*`
-        // 5. Pointers to types with refrenced memory addresses (e.g. variants, generics, arrays, etc.)
+        // 3. Pointers to variable names that start with `*`
+        // 4. Pointers to types with refrenced memory addresses (e.g. variants, generics, arrays, etc.)
         (matches!(child_variable.name, VariableName::Named(ref var_name) if var_name.starts_with('*'))
                 && matches!(parent_variable.role, VariantRole::VariantPart(_)))
             || matches!(&parent_variable.type_name, VariableType::Pointer(Some(pointer_name)) if pointer_name.starts_with('*'))
             || (matches!(&parent_variable.type_name, VariableType::Pointer(_))
                 && (matches!(child_variable.type_name, VariableType::Base(_))
-                    || matches!(child_variable.type_name, VariableType::Struct(ref type_name) if type_name.starts_with("&str"))
```

Comment:
I could see how this kills some other functionality, as I am sure the is_pointer() terminology was chosen after C like languages and uses the term for references too.

I agree though that straightening terminology makes sense though.

Reply:
I just can't find the case where "pointer whose child is &str" makes a difference. I find the whole `is_pointer` a bit arbitrary, especially with the logic assuming Rust for no good reason. I did decide to revert this change, though, because if it makes no difference, it doesn't hurt either.